### PR TITLE
Fix allow_blank supporting DateTime/Date

### PR DESCRIPTION
**Pull request by u2**
#915

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,4 +7,5 @@ group :development, :test do
   gem 'guard'
   gem 'guard-rspec'
   gem 'guard-rubocop'
+  gem 'byebug'
```
Comment (kuraga) [inline Gemfile]: Remove it, please.

**Comment by dblock** [comment]
Needs CHANGELOG, too, please.

File: Gemfile (hunk 1 above)
Comment (u2) [inline Gemfile]: Thank you.I forgot it.

**Comment by kuraga** [comment]
Hm... Is `val == ""` dangerous, isn't?

Can `val` be a non-`String` here?
If can then `"" == val` maybe?
Or `val.empty?` with some additional conditions?

Which values of `val` are possible?

**Comment by u2** [comment]
Before I wrote `val.is_a?(String) && val.empty?`,but I thought it's the same meaning of `val == ""`.
Yes, you are right.This val can be a non-`String`.I will change it asap.

File: lib/grape/validations/validators/coerce.rb [hunk 2]
```diff
@@ -29,9 +29,12 @@ def _valid_single_type?(klass, val)
         # allow nil, to ignore when a parameter is absent
         return true if val.nil?
         if klass == Virtus::Attribute::Boolean
-          val.is_a?(TrueClass) || val.is_a?(FalseClass)
+          val.is_a?(TrueClass) || val.is_a?(FalseClass) || val == ''
```
Comment (kuraga) [inline lib/grape/validations/validators/coerce.rb]: @u2 And here, too?

Comment (u2) [inline lib/grape/validations/validators/coerce.rb]: sorry..

**Comment by dblock** [comment]
Merging, thanks.

**Pull request merged**